qemu/tests/qemu_guest_agent.py

```diff
@@ -3697,8 +3697,9 @@ def _result_check(rsult_qga, rsult_guest):
         _result_check(kernel_release_qga, kernel_release_guest)
 
         error_context.context("Check variant and machine type.", logging.info)
-        if self.params.get("os_variant", "") != 'rhel8':
-            # for rhel8+ there is no variant info
+        main_os_variant = int(self.params.get("os_variant", "")[4:])
+        if main_os_variant < 8:
```

Comment:
```suggestion
        rhel_version =  re.search('rhel(\d)', self.params.get("os_variant", ""))
        if rhel_version and rhel_version.group(1) < '8':
```
In case the os might not be RHEL.

Reply:
@vivianQizhu 
(1/1) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (91.37 s)

Reply:
It should be 'not rhel_version' I bet, because if the guest is windows so that the 'rhel_version' will be 'None'.
rhel_version =  re.search('rhel(\d+)', self.params.get("os_variant", ""))
        if not rhel_version and rhel_version.group(1) < '8':

Reply:
> It should be 'not rhel_version' I bet, because if the guest is windows so that the 'rhel_version' will be 'None'.
> rhel_version = re.search('rhel(\d+)', self.params.get("os_variant", ""))
> if not rhel_version and rhel_version.group(1) < '8':

How do you consider other linux distribution, do you want to test them all or not test them all? For `if rhel_version` I suppose we do not test other linux distribution for this step.

And, `if not rhel_version and rhel_version.group(1) < '8':` might not be fulfilled forever, I guess you mean `or` instead of `and`.
Considering windows together,
if you execute the test for all the other linux distribution:
 `if not rhel_version or rhel_version.group(1) < '8':`
if you do not execute the test for the other linux distribution:
 `if (rhel_version and rhel_version.group(1) < '8') or os_type == "windows":`
Remember we need to consider fedora at least, and I guess new distribution might have same situation on rhel, so how about this:
```
variant_qga = os_info_qga.get("variant", "")
if variant_qga:
    variant_id_qga = os_info_qga["variant-id"]
    ...
```

Reply:
> > It should be 'not rhel_version' I bet, because if the guest is windows so that the 'rhel_version' will be 'None'.
> > rhel_version = re.search('rhel(\d+)', self.params.get("os_variant", ""))
> > if not rhel_version and rhel_version.group(1) < '8':
> 
> How do you consider other linux distribution, do you want to test them all or not test them all? For `if rhel_version` I suppose we do not test other linux distribution for this step.
> 
> And, `if not rhel_version and rhel_version.group(1) < '8':` might not be fulfilled forever, I guess you mean `or` instead of `and`.
> Considering windows together,
> if you execute the test for all the other linux distribution:
> `if not rhel_version or rhel_version.group(1) < '8':`
> if you do not execute the test for the other linux distribution:
> `if (rhel_version and rhel_version.group(1) < '8') or os_type == "windows":`
> Remember we need to consider fedora at least, and I guess new distribution might have same situation on rhel, so how about this:
> 
> ```
> variant_qga = os_info_qga.get("variant", "")
> if variant_qga:
>     variant_id_qga = os_info_qga["variant-id"]
>     ...
> ```
For the first question I don't have plan to test other linux distribution, and consider all you said,  I prefer this kind of change as you wrote at the end. it's more clean and efficient.  Here are the results with your suggestions. thanks @vivianQizhu 
(1/2) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (93.40 s)
 (2/2) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (47.78 s)

Reply:
> For the first question I don't have plan to test other linux distribution, and consider all you said, I prefer this kind of 

@6-dehan Maybe I did not express it clearly, I am not talking about your test plan, we are the main user of this project but not the only one, we need to consider other users, our test cases are supposed to support other linux distribution, especially fedora.